backend/src/exercise/exercise-wrapper.ts

```diff
@@ -224,6 +246,83 @@ export class ExerciseWrapper extends NormalType<
         super(databaseService);
     }
 
+    /**
+     * @param file A **valid** import file
+     */
+    static async importFromFile(
+        databaseService: DatabaseService,
+        file: StateExport,
+        exerciseIds: ExerciseIds
+    ): Promise<ExerciseWrapper> {
+        const importOperations = async (manager: EntityManager | undefined) => {
+            let exercise = new ExerciseWrapper(
+                exerciseIds.participantId,
+                exerciseIds.trainerId,
+                [],
+                databaseService,
+                ExerciseState.currentStateVersion,
+                file.history?.initialState ?? file.currentState,
+                file.currentState
+            );
+            const actions: ActionWrapper[] = [];
+            for (const action of file.history?.actionHistory ?? []) {
+                // Make sure to create the actions in order
+                actions.push(
+                    new ActionWrapper(
+                        databaseService,
+                        action,
+                        exercise.emitterId,
+                        exercise
+                    )
+                );
+            }
+            exercise.temporaryActionHistory.push(...actions);
+            let exerciseEntity: ExerciseWrapperEntity;
+            if (manager !== undefined) {
+                exerciseEntity = await exercise.save(manager);
+            }
+            // Update exercise & reload it from the database
+            const migrationResult = await migrateTo(
+                ExerciseState.currentStateVersion,
+                exercise.stateVersion,
+                manager !== undefined ? exerciseEntity!.id : exercise,
+                manager
+            );
+            exercise =
+                manager !== undefined
+                    ? ExerciseWrapper.createFromEntity(
+                          await databaseService.exerciseWrapperService.getFindById(
+                              exerciseEntity!.id
+                          )(manager),
+                          databaseService
+                      )
+                    : (migrationResult as ExerciseWrapper);
```

Comment:
Refactor (only one `manager !== undefined` check, no side effect in migrate to (or at least a comment).